samples/algorithms/repeat-until-success/RepeatUntilSuccess.qs

```diff
@@ -0,0 +1,222 @@
+        }
+    }
+
+    /// # Summary
+    /// Apply Rz(arctan(2)) on qubits using repeat until success algorithm.
+    ///
+    /// # Input
+    /// ## inputBasis
+    /// Pauli basis to prepare input qubit in
+    /// ## inputValue
+    /// Boolean value for input qubit (true maps to One, false maps to Zero)
+    /// ## limit
+    /// Integer limit to number of repeats of circuit
+    /// ## auxiliary
+    /// Auxiliary qubit
+    /// ## resource
+    /// Resource qubit
+    /// ## target
+    /// Target qubit
+    ///
+    /// # Output
+    /// Tuple of (success, numIter) where success = false if the number of 
+    /// iterations (numIter) exceeds the input <limit>
+    operation ApplyRzArcTan2(
+        inputBasis: Pauli,
+        inputValue: Bool,
+        limit: Int,
+        auxiliary: Qubit,
+        resource: Qubit,
+        target: Qubit
+    )
+    : (Bool, Int) {
+        // Initialize results to One by default.
+        mutable done = false;
+        mutable success = false;
+        mutable numIter = 0;
+
+        repeat {
+            // Assert valid starting states for all qubits
+            AssertMeasurement([PauliX], [auxiliary], Zero,
+             "Auxiliary qubit is not in |+⟩ state.");
+            AssertMeasurement([PauliX], [resource], Zero,
+             "Resource qubit is not in |+⟩ state.");
+            AssertQubitIsInState(target, inputBasis, inputValue);
+
+            // Run Part 1 of the program.
+            let result1 = ApplyAndMeasurePart1(auxiliary, resource);
+            // We'll only run Part 2 if Part 1 returns Zero.
+            // Otherwise, we'll skip and rerun Part 1 again.
+            if (result1 == Zero) { //|0+⟩
+                let result2 = ApplyAndMeasurePart2(resource, target);
+                if (result2 == Zero) { //|00⟩
+                    set success = true;
+                } else { //|01⟩
+                    Z(resource); // Reset resource from |-⟩ to |+⟩
+                    Adjoint Z(target); // Correct effective Z rotation on target
+                }
+            } else { //|1+⟩
+                // Set auxiliary and resource qubit back to |+⟩
+                Z(auxiliary);
+                Reset(resource);
+                H(resource);
+            }
+            set done = success or (numIter >= limit);
+            set numIter = numIter + 1;
+        }
+        until (done);
+        return (success, numIter);
+    }
+
+    /// # Summary
+    /// Initialize axiliary and resource qubits in |+⟩, target in |0⟩ or |1⟩.
+    ///
+    /// # Input
+    /// ## inputBasis
+    /// Pauli basis to prepare input qubit in
+    /// ## inputValue
+    /// Boolean value for input qubit (true maps to One, false maps to Zero)
+    /// ## limit
+    /// Integer limit to number of repeats of circuit
+    /// ## auxiliary
+    /// Auxiliary qubit
+    /// ## resource
+    /// Resource qubit
+    /// ## target
+    /// Target qubit
+    operation InitializeQubits(
+        inputBasis: Pauli,
```

Comment:
[nit] Elsewhere you've used spaces around `:`s, it may be good to do so here as well.